src/analysis/retail/monk/mistweaver/modules/spells/SheilunsGift.tsx

```diff
@@ -22,21 +23,29 @@ class SheilunsGift extends Analyzer {
   cloudsLostSinceLastCast: number = 0;
   curClouds: number = 0;
   overhealing: number = 0;
+  legacyOfWisdomActive: boolean = false;
 
   constructor(options: Options) {
     super(options);
     this.active = this.selectedCombatant.hasTalent(TALENTS_MONK.SHEILUNS_GIFT_TALENT);
     if (!this.active) {
       return;
     }
+    this.legacyOfWisdomActive = this.selectedCombatant.hasTalent(
+      TALENTS_MONK.LEGACY_OF_WISDOM_TALENT,
+    );
     this.addEventListener(
       Events.cast.by(SELECTED_PLAYER).spell(TALENTS_MONK.SHEILUNS_GIFT_TALENT),
       this.onCast,
     );
-    this.addEventListener(
-      Events.heal.by(SELECTED_PLAYER).spell(TALENTS_MONK.SHEILUNS_GIFT_TALENT),
-      this.onHeal,
-    );
+
```

Comment:
Add a comment on why you do this, requires some reasoning about the code to see it

Reply:
I don't know why it appears I like to overcomplicate things - removed in favor of just using the onCast handler for everything